### DISCUSSION TITLE
exclusion of an agent and for specific rules

### DISCUSSION THREAD

**Discussion by iptwins** [wazuh/wazuh#16883] [Q&A]
Hello, I am looking for a solution to a problem I am experiencing, but despite the help of the manual, I can't find my solution.

I have a Wazuh server and about ten agents. (all under debian or ubuntu)
I'm looking for a solution to a problem I'm having, but despite the help of the manual, I can't find my solution: I have a Wazuh server and about ten agents.

I have a Wazuh server and ten agents. for one of these agents, I have a lot of false positives for which I receive email alerts. so I want to disable the sending of email alerts for a specific agent only for specific rules as well.

exemple:
agent: 192.168.1.2
rules: 50180 and 31152

Thank you for your help.

**Comment by mdiego92** [comment]
Hello @iptwins,

For that, you could use the hostname parameter with the negate function. Let me share an example of the 50180 rule with you.
In your local_rules.xml or your custom rule file, please add the following rule:

```
  <rule id="50180" level="10" frequency="8" timeframe="120" ignore="60" overwrite="yes">
    <if_matched_sid>50125</if_matched_sid>
    <hostname negate="yes">192.168.1.2$</hostname>
    <description>MySQL: Multiple errors.</description>
    <mitre>
      <id>T1499</id>
    </mitre>
    <group>service_availability,pci_dss_10.6.1,gpg13_4.3,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,tsc_CC7.2,tsc_CC7.3,</group>
  </rule>
```

Documentation for reference: https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#hostname

You can replicate this procedure with any rule you may need.
Please let me know if it worked for you!

**Reply by iptwins** [reply]
Hello @mdiego92 
Thank you for your reply.

I did a test but it doesn't seem to work, I keep getting the email alerts.

Here is what I put in the file /var/osse/etc/rules/local_rules.xml :


```
<rule id="31151" level="10" frequency="14" timeframe="90" ignore="60" overwrite="yes">
        <if_matched_sid>31101</if_matched_sid>
        <hostname negate="yes">192.168.1.2</hostname>
        <description>Multiple web server 400 error codes </description>
        <description>from same source ip.</description>
        <mitre>
         <id>T1595.002</id>
        </mitre>
        <group>web_scan,recon,pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
</rule>
```
I have of course restarted wazuh-manager.

any ideas ? thanks.

**Reply by mdiego92** [reply]
Hello @iptwins,

Could you please confirm if the IP address is the hostname set up for that agent?
Another way would be using the location parameter as per our documentation: https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#location

Please let me know if this worked for you!

**Comment by iptwins** [comment]
Hello @mdiego92 ,

Sorry I haven't replied in a while ...
I had been busy with other things and now I'm back on the subject because my alerts continue...
Further to your last message, I corrected the rule, because I was using the IP address whereas the field indicates "hostname" which is not the same thing ...
Once corrected and do a restart of wazuh, I no longer have the alerts! great, thank you very much 😊

I have another question on the same subject but more complex...
Is it possible to filter alerts according to conditions ?

Example:
hostname: mytest.exemple.com
Rules: 31151
Level : 10
Source ip (client) : 192.168.1.10
Part of log : POST /page.php?

In fact, I've identified that alerts that meet the above conditions are false positives.
Is it possible to filter only the alerts that would have all the conditions of the example above ?

Thank you for your help.

**Comment by mdiego92** [comment]
Hello @iptwins,

In case you want to set a specific alert level for those conditions, or filter out (setting it to level 0) you can do it with something like this:

```
<rule id="100001" level="0">
        <if_matched_sid>31151</if_matched_sid>
        <hostname>mytest.exemple.com</hostname>
        <srcip>192.168.1.10</srcip>
        <match>POST /page.php?</match>
        <description>Silencing false positive. </description>
</rule>
```

Documentation for reference: https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html

Thank you very much for the feedback!
